inspire_disambiguation/api.py

```diff
@@ -21,127 +21,162 @@
-        pairs_size = current_app.config['DISAMBIGUATION_SAMPLED_PAIRS_SIZE']
-        for pair in sample_signature_pairs(signatures_path, clusters_path, pairs_size):
-            fd.write(json.dumps(pair) + '\n')
-
-
-def save_publications():
-    """Save publications to disk.
-
-    Saves a file to disk called (by default) ``publications.jsonl``, which
-    contains one line per record in INSPIRE with information that will be
-    useful for ``BEARD`` during training and prediction.
+    Args:
+        load_data_path (str): Full path to training data for ethnicity estimator.
+        save_model_path (str): Full path where trained ethnicity model will be saved.
     """
-    with open_file_in_folder(current_app.config['DISAMBIGUATION_PUBLICATIONS_PATH'], 'w') as fd:
-        for publication in get_all_publications():
-            fd.write(json.dumps(publication) + '\n')
-
-
-def train_and_save_ethnicity_model():
-    """Train the ethnicity estimator model and save it to disk."""
     estimator = EthnicityEstimator()
-    estimator.load_data(current_app.config['DISAMBIGUATION_ETHNICITY_DATA_PATH'])
+    estimator.load_data(load_data_path)
+    LOGGER.info("Training EthnicityEstimator. May take a while...")
     estimator.fit()
-    estimator.save_model(current_app.config['DISAMBIGUATION_ETHNICITY_MODEL_PATH'])
+    estimator.save_model(save_model_path)
 
 
-def train_and_save_distance_model():
-    """Train the distance estimator model and save it to disk."""
-    ethnicity_estimator = EthnicityEstimator()
-    ethnicity_estimator.load_model(current_app.config['DISAMBIGUATION_ETHNICITY_MODEL_PATH'])
+def train_and_save_distance_model(
+    ethnicity_model_path, save_distance_model_path, sampled_pairs_size
+):
+    """Train the distance estimator model and save it to disk.
 
-    distance_estimator = DistanceEstimator(ethnicity_estimator)
-    distance_estimator.load_data(
-        current_app.config['DISAMBIGUATION_CURATED_SIGNATURES_PATH'],
-        current_app.config['DISAMBIGUATION_SAMPLED_PAIRS_PATH'],
-        current_app.config['DISAMBIGUATION_SAMPLED_PAIRS_SIZE'],
-        current_app.config['DISAMBIGUATION_PUBLICATIONS_PATH'],
+    Args:
+        ethnicity_model_path (str): Full path where ethnicity model is saved.
+        save_distance_model_path (str): Full path where trained distance model
+            will be saved.
+        sampled_pairs_size (int): Number of pairs to be generated for the training.
+            Note:
+                Must be multiple of 12.
+    """
+    LOGGER.info("Pulling training data from ES")
+    curated_signatures = get_signatures(only_curated=True)
+    input_clusters = get_input_clusters(curated_signatures)
+    LOGGER.info(
+        "Preparing %s pairs from sampled data for training.", sampled_pairs_size
     )
+    pairs = [
+        pair
+        for pair in sample_signature_pairs(
+            curated_signatures, input_clusters, sampled_pairs_size
+        )
+    ]
```

Comment:
maybe it's clearer with `list(sample_signature_pairs(curated_signatures, input_clusters, sampled_pairs_size))`

Reply:
IMO current version is more clear, but what @monaawi thinks?